core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -545,6 +546,12 @@ class RDDSuite extends FunSuite with SharedSparkContext {
     assert(sortedTopK === nums.sorted(ord).take(5))
   }
 
+  test("isEmpty") {
+    assert(sc.emptyRDD.isEmpty())
+    assert(sc.parallelize(Seq[Int]()).isEmpty())
+    assert(!sc.parallelize(Seq(1)).isEmpty())
```

Comment:
I don't think this tests the case where there are multiple partitions but no data in any of the partitions.  Maybe add something like

`assert(sc.parallelize(Seq(1,2,3), 3).filter(_ < 0).isEmpty())`